grug/types/src/coin_pair.rs

```diff
@@ -160,6 +160,63 @@ impl CoinPair {
             },
         ]))
     }
+
+    /// Return true if the coin pair has the given denom.
+    pub fn has(&self, denom: &Denom) -> bool {
+        self.first().denom == denom || self.second().denom == denom
+    }
+
+    /// Return the amount of the given denom in the coin pair.
+    pub fn amount_of(&self, denom: &Denom) -> StdResult<Uint128> {
+        if self.first().denom == denom {
+            Ok(*self.first().amount)
+        } else if self.second().denom == denom {
+            Ok(*self.second().amount)
+        } else {
+            Err(StdError::invalid_coins(format!(
+                "coin pair {:?} doesn't have denom: {}",
+                self, denom
+            )))
```

Comment:
Change

```rust
format!("coin pair {:?} doesn't have denom: {}", self, denom)
```

to

```rust
format!("coin pair {self:?} doesn't have denom: {denom}")
```

Same with the two other `format!` calls below.

Reply:
fixed